jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8325,13 +8325,145 @@ public String indent(int n) {
 /*[ENDIF] Java12 */
 
 /*[IF Java13]*/
-	/*
+	/**
 	 * Determine if current String object is LATIN1.
 	 *
 	 * @return true if it is LATIN1, otherwise false.
 	 */
 	boolean isLatin1() {
 		return LATIN1 == coder();
 	}
+
+	/**
+	 * Format the string using this string as the format with supplied args.
+	 *
+	 * @param args
+	 *          the format arguments to use
+	 *
+	 * @return the formatted result
+	 *
+	 * @see #format(String, Object...)
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String formatted(Object... args) {
+		return String.format(this, args);
+	}
+
+	/**
+	 * Removes the minimum indentation from the beginning of each line and
+	 * removes the trailing spaces in every line from the string
+	 *
+	 * @return this string with incidental whitespaces removed from every line
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String stripIndent() {
+		Iterator<String> iter = lines().iterator();
+		StringBuilder builder = new StringBuilder();
+		int min = Integer.MAX_VALUE;
+
+		while (iter.hasNext()) {
+			String line = iter.next();
+
+			if ((!line.isBlank()) || (!iter.hasNext())) {
+				int count = 0;
+				while ((count < line.length())
+					&& (Character.isWhitespace(line.charAt(count))
+					&& (count < min))
+				) {
+					count++;
+				}
+				min = (count < min ? count : min);
+			}
+
+			if (line.isBlank()) {
+				builder.append("");
+			} else {
+				builder.append(line.stripTrailing());
+			}
+			if (iter.hasNext()) {
+				builder.append("\n");
+			}
+		}
+
+		return builder.toString().indent(-min);
+	}
+
+	/**
+	 * Translate the escape sequences in this string as if in a string literal
+	 *
+	 * @return result string after translation
+	 *
+	 * @throws IllegalArgumentException
+	 *          If invalid escape sequence is detected
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String translateEscapes() {
+		StringBuilder builder = new StringBuilder();
+		char[] charArray = toCharArray();
+		int index = 0;
+		int strLength = length();
+		while (index < strLength) {
+			if (charArray[index] == '\\') {
+				if (index < strLength - 1) {
+					/*[MSG "K0D00", "Invalid escape sequence detected: {0}"]*/
+					throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0D00", "\\")); //$NON-NLS-1$ //$NON-NLS-2$
+				}
+				index++;
+				int octal = 0;
+				switch (charArray[index]) {
+					case 'b':
+						builder.append('\b');
+						break;
+					case 't':
+						builder.append('\t');
+						break;
+					case 'n':
+						builder.append('\n');
+						break;
+					case 'f':
+						builder.append('\f');
+						break;
+					case 'r':
+						builder.append('\r');
+						break;
+					case '\"':
+					case '\'':
+					case '\\':
+						builder.append(charArray[index]);
+						break;
+					case '0':
+					case '1':
+					case '2':
+					case '3':
+						octal = charArray[index] - '0';
```

Comment:
This needs to be fixed to handle longer escape sequences, e.g. `\040`.

Reply:
for octal escapes, the value allowed are from `000` to `377`. Values starting with `0 - 3` the max length is 3 while the rest only have max length of 2. longer escape sequence will fall through to the `4-7` cases.

Reply:
See [1] which describes escape sequences as `\` followed by one, two or three octal digits (the first of three must be 0-3). Support for unicode escapes (see [2]) is also missing.

[1] https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-OctalEscape
[2] https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.3

Reply:
unicode escape is not supported by this API see: [1],
the intensional fall-through covers all octal escape sequences

[1] http://cr.openjdk.java.net/~iris/se/13/build/latest/api/java.base/java/lang/String.html#translateEscapes()